Classes/Domain/NodeTypesProviderInterface.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/*
+ * This file is part of the Neos.Neos.Ui package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\Neos\Ui\Domain;
+
+/**
+ * @internal
```

Comment:
your commit message expresses really well the definition of the roles and id very much like to have that as a doc comment here ^^

> Retrieves information about node type roles and
node type groups.
>
> Roles are a dedicated UI-concept that is meant to
distinguish between document, content and collection nodes. Groups refer
the grouping of node types in the creation dialog.

Additionally i would like to propose a different more obvious and specific naming like `NodeTypeGroupsAndRolesProviderInterface`